src/Processors/Sources/MongoDBSource.cpp

```diff
@@ -196,12 +252,14 @@ namespace
         }
     }
 
+    void insertDefaultValue(IColumn & column, const IColumn & sample_column) { column.insertFrom(sample_column, 0); }
+
     void insertValue(
         IColumn & column,
+        const IDataType & data_type,
         const ValueType type,
         const Poco::MongoDB::Element & value,
         const std::string & name,
```

Comment:
Really interesting why do we need this `ValueType` at all. I would just remove it completely and use only our IDataType